azblob/zt_policy_request_id_test.go

```diff
@@ -0,0 +1,101 @@
+package azblob
+
+import (
+	"context"
+	"errors"
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	chk "gopkg.in/check.v1"
+	"net/http"
+	"net/url"
+)
+
+type requestIDTestScenario int
+
+const (
+	// Testing scenarios for echoing Client Request ID
+	clientRequestIDMissing requestIDTestScenario = 1
+	clientRequestIDError   requestIDTestScenario = 2
```

Comment:
The name `clientRequestIDError` is a bit confusing, since the error happens somewhere else, and not on the ID.